docs/articles/documentation/extending-testcafe/creating-custom-reporter/index.md

```diff
@@ -0,0 +1,143 @@
+
+## Generating Reporter Project
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-reporter` using [npm](https://www.npmjs.com/) (we assume you have pre-installed [node.js](https://nodejs.org/)).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-reporter
+```
+
+Create a new directory and go into it.
+
+```bash
+mkdir MyReporter
+cd MyReporter
+```
+
+This directory is where the generator will place your scaffolded project files.
+
+Then run the reporter generator to create a new reporter project:
+
+```bash
+yo testcafe-reporter
+```
+
+The generator will provide optional settings to customize your reporter.
+You need to specify the reporter name, your GitHub username, the URL of your website,
+and select whether you want to implement a custom error decorator for the reporter
+(you will do this later by using the [createErrorDecorator](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/#createerrordecorator) method).
+
+Yeoman will automatically scaffold out your reporter, install the required dependencies,
+and pull in several useful [Gulp tasks](#workflow-tasks) for your workflow.
+
+## Implementing Reporter
+
+Since the reporter has been scaffolded out, go to the reporter directory and open the `src/index.js` file.
+
+You have to implement four methods of the [ReporterPluginHost](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/) object providing access to your reporter.
+
+```js
+reportTaskStart (/* startTime, userAgents, testCount */) {
+    throw new Error('Not implemented');
+},
+
+reportFixtureStart (/* name, path */) {
+    throw new Error('Not implemented');
+},
+
+reportTestDone (/* name, errs, durationMs, unstable, screenshotPath */) {
+    throw new Error('Not implemented');
+},
+
+reportTaskDone (/* endTime, passed */) {
+    throw new Error('Not implemented');
+}
+```
+
+These methods should output the desired information at certain moments during test run: on task start, on fixture start, on test finish, on task finish.
+
+All the required data is provided to these methods through their parameters.
+To know what data you can use in the methods, refer to the test data at `/generator-testcafe-reporter/generators/app/templates/test/data/test-calls.js`.
+
+Another way to figure out what data and how it can be used in reports is to explore an existing reporter,
+say, the *list* reporter at [https://github.com/DevExpress/testcafe-reporter-list](https://github.com/DevExpress/testcafe-reporter-list).
+You can either run the `preview` Gulp task to view the reporter output in your terminal, or find the etalon report at `/test/data/report-without-colors`.
+
+In the `src/index.js` file, you may also need to implement the [createErrorDecorator](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/#createerrordecorator) method to create a custom error decorator.
```

Comment:
Error decorator is some really advanced stuff, really rarely used. We should describe it only once we will finish with the main functionality.